docs/git-convention.md

```diff
@@ -1,10 +1,162 @@
-# 📌 Git Branch Naming Convention (Aligned with Angular Guideline)
+# CommitLint and Git Branch Naming Convention (Aligned with Angular Guideline)
 
-# 📖 Introduction
+To maintain a clean, structured commit history and optimize team collaboration, we adhere to the Angular Conventional Commits standard for both commit messages and Git branch naming. This ensures every change type is immediately recognizable and supports automation for changelog generation, semantic versioning, and streamlined release processes.
 
-### To maintain a clean commit history and improve team collaboration, we follow Angular Conventional Commits in our Git branch naming. This approach makes it easy to identify change types and automate release processes.
+In addition, we enforce these standards using CommitLint, ensuring that all commit messages conform to the defined rules before they are accepted into the repository.
 
-# 🚀 Branch Naming Format
+This project employs both GitHub Actions and a local pre-commit pipeline to validate commit messages, enforce branch naming conventions, and maintain repository integrity throughout the development workflow.
+
+## Local pipeline
+
+The local pipeline is managed via husky
+
+1. The pre-commit requirements are:
+   - linting on the staged files passes
+2. The pre-push requirements are:
+   - All tests pass
+   - Test coverage is met
+
+## Contributing Workflow
+
+To contribute to this repository, follow these steps:
+
+1. **Fork the Repository**
+
+   - Go to the main repository page on GitHub.
+   - Click the **Fork** button (top-right corner).
+   - This creates a copy of the repository under your GitHub account.
+
+2. **Clone Your Fork**
+
+   ```bash
+   git clone https://github.com/<your-username>/<repo-name>.git
+   cd <repo-name>
+   ```
+
+3. **Create a Feature Branch**
+
+   ```bash
+   git checkout -b feat/my-new-feature
+   ```
+
+4. **Make Your Changes**
+
+   - Follow the commit conventions outlined below.
+   - Ensure all tests pass before committing.
+
+5. **Push to Your Fork**
+
+   ```bash
+   git push origin feat/my-new-feature
+   ```
+
+6. **Open a Pull Request**
+   - Go to the main repository on GitHub.
+   - Click **New Pull Request**.
+   - Select your fork and branch, then create the PR against the `main` branch of the main repo.
```

Comment:
Not sure if we want all PRs to point to `main` by default. May be better to have them point to `develop` and make a note that we will release changes from `develop` to `main` periodically following our release cycles